File: lists/concat2.js


const cat = (a1, a2) => {
    let result = [];

    for (let x in a1) {
        for (let y in a2) {
            let temp = a1[x] + a2[y];
            result.push(temp);
        }
    }
    return result;
}

let list1 = ["Hello ", "take "];
let list2 = ["Dear", "Sir"];

console.log(cat(list1, list2));

File: lists/concat2.py


def cat(a1, a2):
    result = []

    for x in a1:
        for y in a2:
            temp = x + y 
            result.append(temp)

    return result 

list1 = ["Hello ", "take "]
list2 = ["Dear", "Sir"]

print(cat(list1, list2))